contracts/staking/P12MineUpgradeable.sol

```diff
@@ -458,6 +457,9 @@ contract P12MineUpgradeable is
     pool.accP12PerShare = _accP12PerShare;
     pool.period += 1;
     periodTimestamp[pool.lpToken][pool.period] = block.timestamp;
+
+    gaugeController.checkpointGauge(address(pool.lpToken));
```

Comment:
Why change the order here

Reply:
https://github.com/ProjectTwelve/contracts/security/code-scanning/46  
https://github.com/ProjectTwelve/contracts/security/code-scanning/38